src/int64_util.js

```diff
@@ -26,66 +30,92 @@ var POW2_31 = Math.pow(2, 31);
 var POW2_32 = Math.pow(2, 32);
 var POW10_11 = Math.pow(10, 11);
 
-Int64Util.toDecimalString = function(i64) {
-  var b = i64.buffer;
-  var o = i64.offset;
-  if ((!b[o] && !(b[o + 1] & 0xe0)) ||
-      (!~b[o] && !~(b[o + 1] & 0xe0))) {
-    // The magnitude is small enough.
-    return i64.toString();
-  } else {
-    var negative = b[o] & 0x80;
-    if (negative) {
-      // 2's complement
-      var incremented = false;
-      var buffer = new Buffer(8);
-      for (var i = 7; i >= 0; --i) {
-        buffer[i] = (~b[o + i] + (incremented ? 0 : 1)) & 0xff;
-        incremented |= b[o + i];
-      }
-      b = buffer;
+Int64Util.toDecimalString = function (i64)
+{
+    var i64Buffer = i64.buffer;
+    var i64Offset = i64.offset;
+    if ((!i64Buffer[i64Offset] && !(i64Buffer[i64Offset + 1] & 0xe0)) ||
+        (!~i64Buffer[i64Offset] && !~(i64Buffer[i64Offset + 1] & 0xe0)))
+    {
+        // The magnitude is small enough.
+        return i64.toString();
+    }
+    else
```

Comment:
this else is unnecessary (due to the return above)

Reply:
fixed